types/underscore/index.d.mts

```diff
@@ -0,0 +1,5 @@
+// See https://github.com/jashkenas/underscore/blob/master/underscore-node.mjs for reference
+// Note: currently missing `toPath`, `isMap`, `isSet`, `isWeakMap`, `isWeakSet`, `restArguments` and the default is split onto a new line
+export { VERSION, after, every as all, allKeys, some as any, extendOwn as assign, before, bind, bindAll, chain, chunk, clone, map as collect, compact, compose, constant, contains, countBy, create, debounce, defaults, defer, delay, find as detect, difference, rest as drop, each, escape, every, extend, extendOwn, filter, find, findIndex, findKey, findLastIndex, findWhere, first, flatten, reduce as foldl, reduceRight as foldr, each as forEach, functions, get, groupBy, has, first as head, identity, contains as include, contains as includes, indexBy, indexOf, initial, reduce as inject, intersection, invert, invoke, isArguments, isArray, isArrayBuffer, isBoolean, isDataView, isDate, isElement, isEmpty, isEqual, isError, isFinite, isFunction, isMatch, isNaN, isNull, isNumber, isObject, isRegExp, isString, isSymbol, isTypedArray, isUndefined, iteratee, keys, last, lastIndexOf, map, mapObject, matcher, matcher as matches, max, memoize, functions as methods, min, mixin, negate, noop, now, object, omit, once, pairs, partial, partition, pick, pluck, property, propertyOf, random, range, reduce, reduceRight, reject, rest, result, sample, filter as select, shuffle, size, some, sortBy, sortedIndex, rest as tail, first as take, tap, template, templateSettings, throttle, times, toArray, unzip as transpose, unescape, union, uniq, uniq as unique, uniqueId, unzip, values, where, without, wrap, zip } from './index.js';
```

Comment:
This is an ugly piece of duplicate code right here. I hope there is a way we can do without it.

Reply:
Unfortunately not - you can't `export * from` a `export =`'d namespace object, we we need to manually list all the things we want to reexpose at the esm entrypoint.

Reply:
Can't we fix it by an inversion of control? Use exclusively named and default exports in an ESM module first, then go something like `import * as whole_interface` in the module where we do `export = mixin(whole_interface)`? That's [the approach I took in Underscore](https://underscorejs.org/docs/modules/index-default.html), too.

Reply:
Can't use `export =` in an esm format file, which this is, since it has no esm equivalent.

Reply:
Right, but I mean, can we move all the named and default exports from the other file to this one, and then in the *other* file, *only* do `import *` and `export =`?

Reply:
By "move exports" I also mean "move types".

Reply:
I'm pretty sure we can't do that because in cjs the `_` object (which is the module as a whole) is callable - and because you can't `export * from` in a namespace, we'd still need to explicitly relist every export like this.

Reply:
If you forget the existing structure of the DT package for a moment, and take a look at what Underscore proper is *actually* doing ([1](https://underscorejs.org/docs/modules/index.html), [2](https://underscorejs.org/docs/modules/index-default.html), [3](https://underscorejs.org/docs/modules/index-all.html)), don't you think the DT package could just do the same? Number 1 (the module that mirrors `index.js`) could contain all the types internally instead of importing from individual modules.

Reply:
Those docs don't seem to actually reflect what files are exported by underscore and their actual shape. The `import` condition points at a separate file that has explicitly listed reexports, just like this.